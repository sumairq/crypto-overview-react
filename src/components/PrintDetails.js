import React from 'react';
import PropTypes from 'prop-types';

const PrintDetails = ({ data }) => (
  <section className="details-section">
    {Object.entries(data).map((item) => {
      if (typeof item[1] === 'number' && item[1].toString().length < 15 && item[0].toString().length < 15) {
        return (
          <p>
            {item[0]}
            {' '}
            <span>{item[1]}</span>
            {' '}
          </p>
        );
      }
      if (item[1] !== null && typeof item[1] !== 'object' && item[0] !== 'image' && item[0].length < 15) {
        return (
          <p>
            {item[0]}
            {' '}
            <span>{item[1]}</span>
          </p>
        );
      }
      return null;
    })}
  </section>
);
PrintDetails.propTypes = {
  data: PropTypes.shape([PropTypes.object]).isRequired,
};
export default PrintDetails;
