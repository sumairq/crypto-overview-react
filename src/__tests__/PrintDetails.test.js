import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import PrintDetails from '../components/PrintDetails';
import store from '../redux/configureStore';

test('matches voice search component snapshot', () => {
  const coin = {
    id: 'bitcoin',
    symbol: 'btc',
    name: 'Bitcoin',
    current_price: 44260,
    market_cap: 840513708212,
    total_volume: 26316282020,
    high_24h: 44999,
    low_24h: 43721,
    total_supply: 21000000,
    max_supply: 21000000,
    ath: 69045,
    ath_date: '2021-11-10T14:24:11.849Z',
    atl: 67.81,
    atl_date: '2013-07-06T00:00:00.000Z',
    last_updated: '2022-03-25T17:43:30.692Z',
  };

  const tree = renderer.create(
    <Provider store={store}>
      <PrintDetails data={coin} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
