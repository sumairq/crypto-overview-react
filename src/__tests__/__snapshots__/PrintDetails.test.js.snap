// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches voice search component snapshot 1`] = `
<section
  className="details-section"
>
  <p>
    id
     
    <span>
      bitcoin
    </span>
  </p>
  <p>
    symbol
     
    <span>
      btc
    </span>
  </p>
  <p>
    name
     
    <span>
      Bitcoin
    </span>
  </p>
  <p>
    current_price
     
    <span>
      44260
    </span>
     
  </p>
  <p>
    market_cap
     
    <span>
      840513708212
    </span>
     
  </p>
  <p>
    total_volume
     
    <span>
      26316282020
    </span>
     
  </p>
  <p>
    high_24h
     
    <span>
      44999
    </span>
     
  </p>
  <p>
    low_24h
     
    <span>
      43721
    </span>
     
  </p>
  <p>
    total_supply
     
    <span>
      21000000
    </span>
     
  </p>
  <p>
    max_supply
     
    <span>
      21000000
    </span>
     
  </p>
  <p>
    ath
     
    <span>
      69045
    </span>
     
  </p>
  <p>
    ath_date
     
    <span>
      2021-11-10T14:24:11.849Z
    </span>
  </p>
  <p>
    atl
     
    <span>
      67.81
    </span>
     
  </p>
  <p>
    atl_date
     
    <span>
      2013-07-06T00:00:00.000Z
    </span>
  </p>
  <p>
    last_updated
     
    <span>
      2022-03-25T17:43:30.692Z
    </span>
  </p>
</section>
`;
